### ISSUE TITLE
Add administration API to retry dead deliveries or to retry a specific delivery

### ISSUE THREAD

**Issue by gabzim** [wormhole-foundation/relayer-engine#67]
We need a mechanism to go through dead deliveries and retry them. Now this could take different forms. We could have an endpoint to grab the list of dead workflows and return then, then an endpoint to retry a specific workflow, then an endpoint to cleanup workflows (or the entire list) of dead workflows.

Being an administrative endpoint, this needs to seat behind an API Key. Instead of building rbac, or SSO etc, I'd suggest we start off with a single API key as an application secret and authenticate from the `Authentication` header.

**Comment by JoeHowarth** [comment]
Yes I think we should have (1) and (2), not sure we need (3) as a rest endpoint, though we do need to implement a GC job at some point to cleanup old workflows. 

Not too opinionated on the auth mechanism - a single api key seems fine at this point

**Comment by gabzim** [comment]
I was thinking about (3) so that if you retry it and it succeeds, you don't have to see it again when you query (1) since the job to cleanup the dead workflows wouldn't run very often to allow you time to review them.

**Comment by JoeHowarth** [comment]
If the retry succeeds shouldn't that operation remove it from the dead workflows list?

**Comment by gabzim** [comment]
I guess that's ok, I've had use cases in the past where we did multiple actions with failed workflows and when we did that, it would prevent other triggers from doing their thing, but I'm probably overthinking it. Will do that.
